src/components/ColorPicker.tsx

```diff
@@ -1,14 +1,26 @@
 import React, { useState, useEffect, useCallback, useRef } from "react";
+
 import Hue from "./Hue";
 import Saturation from "./Saturation";
-import formatClassName from "../utils/formatClassName";
-import equalColorObjects from "../utils/equalColorObjects";
+
 import styles from "../styles.css";
+import { ColorModel, AnyColor, HSV, ColorPickerBaseProps } from "../types";
+import { equalColorObjects } from "../utils/compare";
+import formatClassName from "../utils/formatClassName";
+
+interface Props extends ColorPickerBaseProps {
+  colorModel: ColorModel<AnyColor>;
+}
 
-const ColorPicker = ({ className, colorModel, color = colorModel.defaultColor, onChange }) => {
+const ColorPicker = ({
+  className,
+  colorModel,
+  color = colorModel.defaultColor,
+  onChange,
+}: Props): JSX.Element => {
```

Comment:
@omgovich This will be something I need your feedback/choice on.

A default rule with the TS-ESLint plugin says that functions should have explicit return types when it can not be absolutely inferred. That would lead to this function needing the `JSX.Element` type while other functions, like in `HexInput.tsx`  don't as the usage of `memo` will immediately narrow down the type. 

Some TS style guides say this is a nuisance and at times the TS team itself says best practice is using inference until you absolutely need to define a type (as the compiler can't work it out itself). Can't link directly, but the small paragraph right above [this link](https://www.typescriptlang.org/docs/handbook/generics.html#working-with-generic-type-variables) says "While type argument inference can be a helpful tool to keep code shorter and more readable, you may need to explicitly pass in the type arguments as we did in the previous example when the compiler fails to infer the type, as may happen in more complex examples."

Pretty much just a stylistic choice. Could just continue as-is, and add a `JSX.Element` or two even if they aren't strictly speaking necessary, or could rely on inference more heavily and disable the rule.

Reply:
Just something I came across. Keeping as-is might just look like an inconsistency, as you spotted with the use of `React.FC`, when in reality it's just there to help the linter along.

Reply:
It seems to me that the code you just commited is fine. We don't need React.FC since we don't use anything it provides (children, defaultProps, etc).

Reply:
We don't need `React.FC`, but would `JSX.Element`. Just to clarify, `HexInput` is perfectly fine for the linter as:

```
const HexInput = (props: Partial<Props>) => {
```

but `ColorPicker` needs:

```
const ColorPicker = ({ ... }: Props): JSX.Element => {
```

Because while `HexInput` is exported like:

```
export default React.memo(HexInput);
```

`ColorPicker` is as:

```
export default ColorPicker;
```

It's no problem to add the extra typing, certainly harmless enough, but might look inconsistent. Thought I'd explain why as you had noted the inconsistency when I used `React.FC` in some places but not others. This is the same "issue" I guess, as the linter will call for some explicit typings in some places but not others.

Reply:
You know what I just thought: we could move memo to ColorPicker.tsx since we use HOC that passes colorModel as a prop. 

The reason why I wrapped the component with memo in each package was becase I used defaultProps that can't be applied to menoized component.

Reply:
Well, it's (potentially) going to be an "issue" across your code base even if you manipulate 1 part into matching the rest is my point.

Reply:
It just seems better to me to write `React.memo` once instead of copying it for each package.
Feels like the right way because it makes the code simpler.